src/AnalysisDriver.h

```diff
@@ -0,0 +1,19 @@
+#ifndef __ANALYSIS_DRIVER_H__
+#define __ANALYSIS_DRIVER_H__
+
+#include "Configuration.h"
+#include "EvalExpressionAnalysis.h"
+#include <memory>
+
+class AnalysisDriver {
+
+  public:
+    AnalysisDriver(const Configuration &configuration);
+    void closure_entry(const SEXP call, const SEXP op, const SEXP rho);
+
+  private:
+    Configuration configuration_;
```

Comment:
shall this be a const ref?

Reply:
I don't like storing references to external objects. Its always tricky because one has to guarantee that the Configuration object's lifetime is >= the lifetime of the objects holding its const reference. Its sometimes easy to overlook this fact or break this constraint and get very puzzling segfaults because the object one is holding a reference to has been destroyed. 

- One way of getting around this limitation is to ensure that Configuration object is lightweight and does not allow any modification of its contents so that all its instances are always in sync. This is the approach I have taken.

- The other approach is to use a `shared_ptr` to reflect the fact that the object is shared by multiple classes and can't be destroyed untill there are no references to it.

Reply:
This make sense. So each instance that needs a configuration gets it as a reference, but copies it and stores its own version. As long as it is reasonably small and immutable it shall be good. The field shall be then:

```cpp
const Configuration configuration_;
```